src/window/update_loop.rs

```diff
@@ -25,7 +26,8 @@ const MAX_ANIMATION_DT: f32 = 1.0 / 120.0;
 pub struct UpdateLoop {
     idle: bool,
     previous_frame_start: Instant,
-    dt: f32,
+    last_dt: f32,
+    frame_dt_avg: NoSumSMA<f64, f64, 10>,
```

Comment:
This is a reasonable change, but should really have a comment explaining why the moving average is useful. Its not clear at a glance coming to this code with even some familiarity

Reply:
Im not sure I understand fully why. Leaving this for Fredizzimo

Reply:
I can add a comment.

A modern rendering pipeline is highly asynchronous, so we can't rely on timing of individual frames to get the frame rate. The cpu times we get could fluctuate very much, even if the frames are shown to the user at an absolute stable rate. And we want to do our calculations based on that rate. 

Averaging the client rates gives the correct result, since the client can't display less or more frames than are actually rendered.